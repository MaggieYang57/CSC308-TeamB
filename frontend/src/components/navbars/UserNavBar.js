import React, { Component } from "react";
import { logout } from "./LogoutFunc";

class UserNavBar extends Component {
  render() {
    return (
      <div>
        <nav
          className="navbar navbar-dark "
          style={{ backgroundColor: "#2c6674" }}
        >
          <div className="navbar-brand">
            <form className="form-inline " style={{ marginLeft: "-11vw" }}>
              <a href="/">
                <img
                  className="nav-link"
                  src="/images/SH.png"
                  style={{ width: 200, height: 53, borderRadius: 0 }}
                />
              </a>
              <a className="nav-link text-light" href="/profile">
                Profile
              </a>
              <a className="nav-link text-light" href="/hikeFinder">
                HikeFinder
              </a>
            </form>
          </div>
          <form
            className="form-inline"
            style={{ marginRight: "2vw" }}
            action="/logout"
          >
            <button
              className="btn btn-outline-light my-2 my-sm-0 m-3 border border-white"
              onClick={logout}
            >
              Logout
            </button>
          </form>
        </nav>
      </div>
    );
  }
}

export default UserNavBar;
