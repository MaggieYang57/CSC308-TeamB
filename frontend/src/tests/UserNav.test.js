import { fireEvent, render, screen } from "@testing-library/react";
import React from 'react';
import UserNavBar from '../components/navbars/UserNavBar';
import renderer from 'react-test-renderer';

test("clicking on HikeFinder takes you to the HikeFinder page", () => {
  render(<UserNavBar />)
  fireEvent.click(screen.getByRole('link', { name: 'Profile'}))
})

it('renders correctly', () => {
  const tree = renderer
    .create(<UserNavBar/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});