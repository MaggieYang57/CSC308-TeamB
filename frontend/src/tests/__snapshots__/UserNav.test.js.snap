// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="navbar navbar-dark "
    style={
      Object {
        "backgroundColor": "#2c6674",
      }
    }
  >
    <div
      className="navbar-brand"
    >
      <form
        className="form-inline "
        style={
          Object {
            "marginLeft": "-11vw",
          }
        }
      >
        <a
          href="/"
        >
          <img
            className="nav-link"
            src="/images/SH.png"
            style={
              Object {
                "borderRadius": 0,
                "height": 53,
                "width": 200,
              }
            }
          />
        </a>
        <a
          className="nav-link text-light"
          href="/profile"
        >
          Profile
        </a>
        <a
          className="nav-link text-light"
          href="/hikeFinder"
        >
          HikeFinder
        </a>
      </form>
    </div>
    <form
      action="/logout"
      className="form-inline"
      style={
        Object {
          "marginRight": "2vw",
        }
      }
    >
      <button
        className="btn btn-outline-light my-2 my-sm-0 m-3 border border-white"
        onClick={[Function]}
      >
        Logout
      </button>
    </form>
  </nav>
</div>
`;
